pilot/pkg/xds/filters/filters.go

```diff
@@ -301,27 +304,70 @@ var (
 	ConnectAuthorityFilter = &hcm.HttpFilter{
 		Name: "connect_authority",
 		ConfigType: &hcm.HttpFilter_TypedConfig{
-			TypedConfig: protoconv.TypedStruct("type.googleapis.com/io.istio.http.connect_authority.Config"),
+			TypedConfig: protoconv.MessageToAny(&sfs.Config{
+				OnRequestHeaders: []*sfsvalue.FilterStateValue{
+					{
+						Key: &sfsvalue.FilterStateValue_ObjectKey{
+							ObjectKey: "envoy.filters.listener.original_dst.local_ip",
```

Comment:
@kyessenov is there a reason this ObjectKey has to be tied to a `well_known_filter_state`?  I noticed I get this [error](https://github.com/envoyproxy/envoy/blame/5228bf8f080358a00f3996d3c3a68f12f5419b1e/source/extensions/filters/common/set_filter_state/filter_config.cc#L22) if I try to set an arbitrary key value.

I don't think this was a requirement in the original `decoder_callbacks_->streamInfo().filterState()->setData(<key>,...)` function in proxy - is there a reason for this regression?

Reply:
It was left out from the initial version. The reason is you need to look up the C++ object type via a key.

Reply:
>initial version

So there are plans to support this in the near term?

Reply:
Yes, string and integer keys were requested.

Reply:
ok, I see the discussion:
https://github.com/envoyproxy/envoy/pull/29844#discussion_r1355812830

thanks for clarifying